core/Symbols.h

```diff
@@ -582,6 +582,18 @@ class Symbol final {
     const InlinedVector<Loc, 2> &sealedLocs(const GlobalState &gs) const;
     TypePtr sealedSubclassesToUnion(const GlobalState &ctx) const;
 
+    // Record a required ancestor for this class of module
+    void recordRequiredAncestor(MutableContext ctx, SymbolRef ancestor, Loc loc);
+
+    // Associate a required ancestor with the loc it's required at
+    struct RequiredAncestor {
+        SymbolRef symbol;
+        Loc loc;
+    };
```

Comment:
I'd add a constructor for `RequiredAncestor` to make working with vectors easier.